easybuild/easyblocks/m/molpro.py

```diff
@@ -218,16 +227,24 @@ def install_step(self):
                     r"directory .* does not exist, try to create [Y]/n\n": '',
                 }
                 run_cmd_qa(cmd, qa=qa, std_qa=stdqa, log_all=True, simple=True)
+
         else:
             if os.path.isfile(self.license_token):
                 run_cmd("make tuning")
 
             super(EB_Molpro, self).install_step()
 
-            # put original LAUNCHER definition back in place in bin/molpro that got installed,
-            # since the value used during installation point to temporary files
-            molpro_path = os.path.join(self.full_prefix, 'bin', 'molpro')
-            apply_regex_substitutions(molpro_path, [(r"^(LAUNCHER\s*=\s*).*$", r"\1 %s" % self.orig_launcher)])
+            destroot = self.full_prefix
+
+        molpro_exe = os.path.join(destroot, 'bin', 'molpro')
+
+        # Make sure that the launcher text we replace (where this is done) is not
+        # the instance of setting LAUNCHER that contains the eval.
+        launchertext = r"^(LAUNCHER\s*=\s*)((?!\"`eval echo \$LAUNCHER` \$MOLPRO_OPTIONS\").)*$"
+        if self.cfg['parallel_launcher'] is not None:
+            apply_regex_substitutions(molpro_exe, [(launchertext, r'\1 "%s"' % self.cfg['parallel_launcher'])])
+        elif self.orig_launcher is not None:
+            apply_regex_substitutions(molpro_exe, [(launchertext, r"\1 %s" % self.orig_launcher)])
```

Comment:
@valtandor while testing this, I noticed that the syntax here is not correct, i.e.:

```
$ grep '^LAUNCHER' $EASYBUILD_PREFIX/software/Molpro/*/*/bin/molpro
LAUNCHER= /path/to/software/impi/5.0.2.044-iccifort-2015.1.133-GCC-4.9.2/bin64/mpiexec -machinefile %h -n %n %x
```

the space after the `=`, that's just not right in a bash script...

This issue was already there in the easyblock however, you didn't introduce it. But we should fix this while we're at it.

This additional change fixes that problem, please review/merge: https://github.com/valtandor/easybuild-easyblocks/pull/4